code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -63,9 +63,7 @@
 			M.client.perspective = MOB_PERSPECTIVE
 
 /obj/structure/closet/proc/moveMob(var/mob/M, var/atom/destination)
-	loc.Exited(M)
-	M.loc = destination
-	loc.Entered(M, ignoreRest = 1)
+	M.forceMove(destination)

```

Comment:
This is uh, meant to be different for a reason. Otherwise you can permastun/slip/etc people by putting a bar of soap in a locker and spamming it open and closed until the end of time.